src/features/machines/publicKeyVerifierMachine.ts

```diff
@@ -0,0 +1,111 @@
+import type { providers } from "near-api-js"
+import type { CodeResult } from "near-api-js/lib/providers/provider"
+import { fromPromise } from "xstate"
+import { settings } from "../../config/settings"
+import type { Transaction } from "../../types/deposit"
+
+export type SendNearTransaction = (
+  tx: Transaction
+) => Promise<{ txHash: string } | null>
+
+export const publicKeyVerifierMachine = fromPromise(
```

Comment:
This isn't a machine per se, but rather an actor implementation. Let's avoid mixing concerns and create an actors folder within the machines directory, as this is confusing to me and I believe it will confuse others as well.

Reply:
It is not mixing concerns. It is co-location, related logic live in one place. They should not be used separately.

Reply:
For now, this isn't a machine because you're only using fromPromise, which is actor logic. It's more like a single piece of logic rather than a complete machine."

Reply:
We literally have the same approach in src/features/machines/queryQuoteMachine.ts